pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -430,7 +437,10 @@ class Timestamp(_Timestamp):
             raise ValueError("Cannot pass a datetime or Timestamp with tzinfo with "
                              "the tz parameter. Use tz_convert instead.")
 
-        ts = convert_to_tsobject(ts_input, tz, unit, 0, 0, nanosecond or 0)
+        if getattr(ts_input, 'fold', None) is not None and fold is None:
+            fold = ts_input.fold
```

Comment:
Try to get fold from input if it wasn't passed.

Reply:
Will need to raise in the case where `fold` is specified in the `ts_input` and in the `Timestamp` constructor and they are different:

e.g. `ts = Timestamp(datetime.datetime(2020, 1, 1, fold=1), fold=0)`

Reply:
@mroeschke 
The problem is that in datetime fold cannot be None, so we would be locking `fold` in for any input that has `fold` of its own.
Then, if we want to raise, it might just be better to keep the default as None and do something like:
```python
if getattr(ts_input, "fold", None) is not None and fold is not None:
    raise ValueError("Cannot set fold for datetime or Timestamp input")
```

Reply:
Gotcha. Okay as is then.